### DISCUSSION TITLE
Nested if/repeat blocks is VSCode

### DISCUSSION THREAD

**Discussion by NevilleDNZ** [alefragnani/vscode-language-pascal#73] [Q&A] [answered]
Love the extension.  Thank you.  I do have a small puzzle that someone might be able to hint and resolve.

I trying to use the ```if else (no begin end)``` snippet and am getting mixed results.

I tab past ```True``` to get to the end of ```then code here␣```...
```
if True then
    then code here␣
else
    else code here
```
Where ␣ indicates where the cursor is.

Then I type in ```repeat```... ... and wait.... no ```repeat ... until()``` snippet is offered.
```
if True then
    repeat␣
else
    else code here
```
I was expecting an offer of the ```repeat``` snippet.  eg.
```
if True then
    repeat
        ␣
    until (True);
else
    else code here

```
But I just got ```repeat``` but **without** the ```until (True)```

The issue seems too be that if a ```primary snippet``` field has "defaulttext" (eg ```${1:defaulttext}```) then the ```secondary snippets``` are not offered.  

I **suspect** that this **may** not be a problem on ```atom```, just ```VSCode```.  (Maybe someone can confirm?)

FYI:  I am mostly up-to-date.
- alefragnani.pascal - Alessandro Fragnani - 134,339 - v9.2.0
- Version: 1.52.1
- Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
- Date: 2020-12-16T16:32:10.090Z
- Electron: 9.3.5
- Chrome: 83.0.4103.122
- Node.js: 12.14.1
- V8: 8.3.110.13-electron.0
- OS: Linux x64 3.10.0-1160.11.1.el7.x86_64

**Comment by alefragnani** [comment, marked as answer]
Hi @NevilleDNZ ,

This is a VS Code behavior, not the extension/snippets. I just tested some _native_ TypeScript snippets, and it does the same thing. And it does not matter if the snippet has `defaulttext` or not, the _secondary_ snippet is not automatically shown.

Otherwise, you can **change** this behavior, if you change a VS Code setting 😬 

1. Open `Preferences / Configure User Settings`
2. Search for `snippet`
3. You will find a setting named `Editor > Suggest: Snippets Prevent Quick Suggestions`
4. Uncheck the checkbox 

Now the _secondary_ snippets will be shown

Hope this helps
